### ISSUE TITLE
Falloff Node

### ISSUE THREAD

**Issue by jonassorgenfrei** [charlesangus/DeepC#96]
This node idea came up at work and already had a quick chat with @falkhofmann about it. 

The goal is to have a spherical fall off node which reduces the alpha based on a falloff ramp and radius parameters.

Inputs: 
- deep 
- camera
- axises to have one or multiple sphere centers

Parameters:
- opaque radius 
- additional falloff radius
- ramp to define the falloff
- toggle if axis input should be used or fall off around camera

-> if points are outside of falloff radius drop the deep samples otherwise multiply alpha
-> the axis shall be able to be animated so the fall of center be move, so it can be easy later to parent the axis to animations 

I thinks this tool sounds straight forward and is not to complicated to implemented with the already existing code as reference. So i think ill be able to implement it. 

If you have any toughts or additional ideas about this tool please write them in the comments to this ticket :)

**Comment by charlesangus** [comment]
I think this would be better handled by a DeepCPremult and/or or DeepCMask
or similar - you can then add arbitrary mattes (including spheres) to the
Deep stream using e.g. DeepCPMatte and then mask by that matte using the
DeepCPremult.

Could also be used to easily correct renders where the data stream is
(incorrectly) unpremultiplied.

Also implies the existence of a DeepCUnPremult, I suppose.

**Comment by jonassorgenfrei** [comment]
Yeah but this would need the Masks to be created in a previous step. 
So the idea would be to create this "virtual" masks on the fly by just plugging in an (pos. animated) axis or multiple ones.
The idea would be to use this to relight fog/volume deep point clouds with point lights and spherical falloffs.

**Comment by charlesangus** [comment]
But if you're using an arbitrary mask already in the stream to do the work,
a) that's more in line with how nuke usually works and b) this node would
benefit from any other deep generator nodes made down the line.

Could also process the matte beforehand this way, e.g. take a sphere matte
and mask it by noise.

I get the appeal of the simplicity of one node to do this one simple thing,
but I feel like this mask node will then start accreting features that
duplicate other functionality - "oh can it just...". And then if that new
functionality would also be useful elsewhere, then you end up either
missing it or copying it.

If the sticking point is axis input, I don't think it would be crazy to add
that to DeepCPMatte.

Thoughts?

**Comment by jonassorgenfrei** [comment]
Okay got it. Totally makes sense and im a fan of more modular structures aswell tho. 

So you would suggest to:
-> enhace the DeepCPMatte Node with an optional (or multiple) axis to input
-> the mask will be stored on the deep data 
-> we add an DeepCMask node, that can mask any channel by any given mask already on deep? 

Please correct me if I got something wrong. 

Does it make sense to add an optional falloff ramp from the center to the DeepCMatte then tho or would you do this in a kind of post process?

**Comment by charlesangus** [comment]
Yeah, that's the thought.

DeepCPMatte already has falloff options, but you can add more if you don't
see the one you like.
